test/src/test/java/hudson/model/ItemGroupMixInTest.java

```diff
@@ -47,4 +55,76 @@
         assertEquals("valid", items.iterator().next().getName());
     }
 
+  /**
+   * This test unit makes sure that if part of the config.xml file is
+   * deleted it will still load everything else inside the folder.
+   * The test unit expects an IOException is thrown, and the one failed
+   * job fails to load.
+   */
+  @Issue("JENKINS-22811")
+  @Test
+  public void xmlFileFailsToLoad() throws Exception {
+    MockFolder folder = r.createFolder("folder");
+    assertNotNull(folder);
+
+    AbstractProject project = folder.createProject(FreeStyleProject.class, "job1");
+    AbstractProject project2 = folder.createProject(FreeStyleProject.class, "job2");
+    AbstractProject project3 = folder.createProject(FreeStyleProject.class, "job3");
+
+    File configFile = project.getConfigFile().getFile();
+
+    List<String> lines = FileUtils.readLines(configFile).subList(0, 5);
+    configFile.delete();
+
+    // Remove half of the config.xml file to make "invalid" or fail to load
+    FileUtils.writeByteArrayToFile(configFile, lines.toString().getBytes());
+    for (int i = lines.size() / 2; i < lines.size(); i++) {
+      FileUtils.writeStringToFile(configFile, lines.get(i), true);
+    }
+
+    // Reload Jenkins.
+    r.jenkins.reload();
+
+    // Folder
+    assertNotNull("Folder failed to load.", folder);
```

Comment:
Right, this is just a local variable which obviously will still be non-null.

Remember that you need to write tests _first_, verifying that they fail (in the expected way), _then_ write the main source fix to make them pass. If you forget and write the fix first, you need to go back and temporarily comment it out while you perfect the test.